app/jobs/consolidated_item_email_job.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+class ConsolidatedItemEmailJob < ApplicationJob
+  # When the job runs, it will check emails that should have been sent within
+  # EMAIL_CHECK_INTERVAL.
+  # This also implies that the job should be configured to run at the same rate so items
+  # don't get missed and duplicate emails don't get sent.
+  EMAIL_CHECK_INTERVAL = 10.minutes
+
+  # Start with opening reminders.
+  def perform
+    generate_opening_email_items.each do |course, course_emails_hash|
+      course_emails_hash.each do |user_email, items_hash|
+        ConsolidatedMailer.opening_reminder_email(course, user_email, items_hash).deliver_later
+      end
+    end
+  end
+
+  private
+
+  # Generates a nested hash with the following structure for opening reminders:
+  # {
+  #   Course_1 => {
```

Comment:
You have to be careful with the documentation - Ruby class constants can actually act as hash keys, but in this case you are taking the string of the Ruby class constant.